### ISSUE TITLE
CHECKSUM does not work if checksum is not supplied with OO interface

### ISSUE THREAD

**Issue by Pascal666** [thrig/File-AtomicWrite#9]
When CHECKSUM is true, write_file appears to work correctly if I do not specify the checksum.  When using new/commit I always get "temporary file SHA1 hexdigest does not match supplied checksum" if I do not supply it.

**Comment by thrig** [comment]
What exact code are you using? E.g. did you copy the code for ->checksum($digest)->commit from the tests or synopsis, or did you write something else?

**Comment by thrig** [comment]
you need to call ->checksum somewhere, like is documented in the SYNOPSIS, or see how the test code does it

**Comment by Pascal666** [comment]
Yes, exactly.  I always get "temporary file SHA1 hexdigest does not match supplied checksum" if I do not call ->checksum.

**Comment by thrig** [comment]
so call ->checksum first, as is documented

**Comment by Pascal666** [comment]
The documentation indicates calling ->checksum is optional:

checksum => sha1 hexdigest
If this option exists, and CHECKSUM is true, the module will not create a Digest::SHA1 hexdigest of the data being written out to disk, but instead will rely on the value passed by the caller.

CHECKSUM => true or false
If true, Digest::SHA1 will be used to checksum the data read back from the disk against the checksum derived from the data written out to the temporary file.

Use the checksum option (or checksum method) to supply a Digest::SHA1 hexdigest checksum. This will spare the module the task of computing the checksum on the data being written.

**Comment by thrig** [comment]
The documentation you cite is mostly for write_file, which is not the OO interface. It may be more productive to use the OO interface as shown in t/020-oo.t as the OO interface does not have the complete data like write_file does, so a checksum method call must be made.
